src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -115,14 +114,15 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
             Type = type;
             Options = options;
 
-            ClassType = GetClassType(
-                type,
-                parentClassType: type,
-                propertyInfo: null,
-                out Type? runtimeType,
-                out Type? elementType,
-                out JsonConverter? converter,
-                options);
+            JsonConverter converter = GetConverter(
+                Type,
+                parentClassType: null, // A ClassInfo never has a "parent" class.
+                propertyInfo: null, // A ClassInfo never has a "parent" property.
+                out Type runtimeType,
+                Options);
+
+            ClassType = converter.ClassType;
+            PolicyProperty = CreatePolicyProperty(Type, runtimeType, converter, Options);
```

Comment:
This didn't change with this PR (PropertyPolicy was always created).

Currently PolicyProperty is always accessed for a given property, so deferring the construction will cause an extra `if` check at runtime instead of doing the creation up-front (which is cached).

Since there's a reference back to the parent, caching would only help for properties on a given type if they're the same Type. PropertyPolicy could pehaps be improved here, but not in scope for this PR anyway.